test/tests.js

```diff
@@ -400,3 +272,118 @@ describe('OneTrust Forwarder', function() {
         });
     });
 });
+
+describe('OneTrust re-initialization tests', function() {
+    before(function() {
+        server.start();
+        server.requests = [];
+    });
+
+    beforeEach(function() {
+        localStorage.clear();
+        window.Optanon = new MockForwarder();
+        server.requests = [];
+        window.mParticleAndroid = null;
+        window.mParticle.isIOS = null;
+        window.mParticle.useCookieStorage = false;
+        mParticle.isDevelopmentMode = false;
+        server.handle = function(request) {
+            request.setResponseHeader('Content-Type', 'application/json');
+            request.receive(
+                200,
+                JSON.stringify({
+                    Store: {},
+                    mpid: 'testMPID',
+                })
+            );
+        };
+    });
+
+    it('should update consent during initialization if consent has changed', function() {
+        var consentGroups = [
+            {
+                jsmap: null,
+                map: 'strictly necessary',
+                maptype: 'ConsentPurposes',
+                value: 'userEditable1',
+            },
+        ];
+
+        // Previous version of kit allowed OnetrustActiveGroups to only be numbers, now they are user editable
+        // The following example includes both examples to allow for maximum flexibility in the test
+        window.OnetrustActiveGroups = ',1,2,4,userEditable1,userEditable2';
+
+        // After initializing the kit once, the user will have the strictly necessary consent associated with it
+        // as true
+        configureOneTrustForwarderAndInit(
+            generateConsentGroupString(consentGroups)
```

Comment:
I was going back and forth on this one. Based on the naming convention, I would assume each of these would be an object, but based on the usage, I see that these are actually stringified objects.

I think it makes things more readable if you move `generateConsentGroupString` into `configureOneTrustForwarderAndInit`, especially since it appears each of the arguments are stringified groups.